samples/matter/common/src/event_triggers/event_triggers.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "event_triggers.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace Nrf::Matter;
+using namespace chip;
+
+CHIP_ERROR TestEventTrigger::RegisterTestEventTrigger(EventTrigger trigger)
+{
+	VerifyOrReturnError(trigger.Callback != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+	VerifyOrReturnError(GetTriggerSlot(trigger.Id) == kInvalidTriggerSlot, CHIP_ERROR_INVALID_ARGUMENT);
+	VerifyOrReturnError(mTriggersMap.FreeSlots() > 0, CHIP_ERROR_NO_MEMORY);
+
+	uint16_t freeSlot = mTriggersMap.GetFirstFreeSlot();
+	VerifyOrReturnError(freeSlot < kMaxEventTriggers, CHIP_ERROR_NO_MEMORY);
+
+	mTriggersMap.Insert(freeSlot, std::move(trigger));
+
+	LOG_INF("Registered new test event: 0x%llx on slot %d", trigger.Id, freeSlot);
+
+	return CHIP_NO_ERROR;
+}
+
+void TestEventTrigger::UnregisterTestEventTrigger(EventTriggerId id)
+{
+	TriggerSlot slot = GetTriggerSlot(id);
+	if (slot == kInvalidTriggerSlot) {
+		return;
+	}
+
+	LOG_INF("Unregistered test event: 0x%llx", id);
+
+	mTriggersMap.Erase(slot);
+}
+
+bool TestEventTrigger::DoesEnableKeyMatch(const chip::ByteSpan &enableKey) const
+{
+	return !mEnableKey.empty() && mEnableKey.data_equal(enableKey);
+}
+
+CHIP_ERROR TestEventTrigger::SetEnableKey(const chip::ByteSpan &newEnableKey)
+{
+	VerifyOrReturnError(newEnableKey.size() == chip::TestEventTriggerDelegate::kEnableKeyLength,
+			    CHIP_ERROR_INVALID_ARGUMENT);
+	return CopySpanToMutableSpan(newEnableKey, mEnableKey);
+}
+
+CHIP_ERROR TestEventTrigger::HandleEventTrigger(uint64_t eventTrigger)
+{
+	/* Check if the Enable Key is set to the "disabled" value. */
+	if (!IsEventTriggerEnabled()) {
+		return CHIP_ERROR_INCORRECT_STATE;
+	}
+
+	TriggerSlot slot = GetTriggerSlot(eventTrigger);
+
+	VerifyOrReturnError(slot != kInvalidTriggerSlot, CHIP_ERROR_INVALID_ARGUMENT);
+	VerifyOrReturnError(mTriggersMap[slot].Callback != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+	return mTriggersMap[slot].Callback(eventTrigger & mTriggersMap[slot].Mask);
+}
+
+uint16_t TestEventTrigger::GetTriggerSlot(uint64_t eventTrigger)
+{
+	for (uint16_t slot = 0; slot < kMaxEventTriggers; slot++) {
```

Comment:
What's actually the value of using `FiniteMap` in this case if you still have to loop over all entries because your _real_ key is `uint64_t` while `FiniteMap` supports only `uint16_t` keys. I think either `FiniteMap` should use keys of any type or it would be easier and more efficient to use a plain array of triggers.

Reply:
Yes, originally the FiniteMap was developed as a dedicated Matter Bridge utility, not with generic usability in mind. So we may try to make it more versatile or use something else here.

Reply:
That's right I wanted to align with the current solution, but I will propose something more efficient :)

Reply:
I added the FiniteDict, to cover our requirements and thoughts, @markaj-nordic @Damian-Nordic what do you think?